python_scripts/01_tabular_data_exploration.py

```diff
@@ -224,81 +224,70 @@
+# in the dataset. Blue points mean low-income and orange points mean
+# high-income. This part of the plot is the same as the bottom-left plot in
 # the pairplot above.
 #
-# What is new in this plot is that we have added the model decision rules as
-# background colors. The background color in each area represents the
-# probability of the class `high-income` as estimated by the model. Values
-# towards 0 (dark blue) indicates that the model predicts `low-income` with a
-# high probability. Values towards 1 (dark orange) indicates that the model
-# predicts `high-income` with a high probability. Values towards 0.5 (white)
-# indicates that the model is not very sure about its prediction.
+# In this plot, we can try to find regions that mainly contains a single class
+# such that we can easily decide what class one should predict. Let's define
+# two such rules that we depict below.
+
+# %%
+import matplotlib.pyplot as plt
+
+ax = sns.scatterplot(
+    x="age", y="hours-per-week", data=adult_census[:n_samples_to_plot],
+    hue="class", alpha=0.5,
+)
+
+age_limit = 27
+plt.axvline(x=age_limit, ymin=0, ymax=1, color="black", linestyle="--")
+
+hours_per_week_limit = 40
+plt.axhline(
+    y=hours_per_week_limit, xmin=0.18, xmax=1, color="black", linestyle="--"
+)
+
+plt.annotate("<=50K", (17, 25), rotation=90, fontsize=35)
+plt.annotate("<=50K", (35, 20), fontsize=35)
+_ = plt.annotate("???", (45, 60), fontsize=35)
+
+# %% [markdown]
+# Looking at the plot, we can define the following rules:
 #
-# Looking at the plot, here is what we can gather:
+# * In the region `age < 27` (left region) the prediction is low-income.
+#   Indeed, there are many blue samples and we cannot see any orange samples.
+# * In the region `age > 27 AND hours-per-week < 40`
+#   (bottom-right region), the prediction is low-income. Indeed, there are
+#   many blue samples and few orange samples.
+# * In the region `age > 27 AND hours-per-week > 40` (top-right region),
+#   we see as many blue samples as orange samples. Indeed, it is complicated
+#   to predict a specific class in this case.
 #
-# * In the region `age < 28.5` (left region) the prediction is `low-income`.
-#   The dark blue color indicates that the model is quite sure about its
-#   prediction.
-# * In the region `age > 28.5 AND hours-per-week < 40.5`
-#   (bottom-right region), the prediction is `low-income`. Note that the blue
-#   is a bit lighter that for the left region which means that the algorithm is
-#   not as certain in this region.
-# * In the region `age > 28.5 AND hours-per-week > 40.5` (top-right region),
-#   the prediction is `low-income`. However the probability of the class
-#   `low-income` is very close to 0.5 which means the model is not sure at all
-#   about its prediction.
+# It is interesting to note that some machine learning models will work
+# similarly to what we did: they are known as decision tree models. Note that
```

Comment:
It would be nice here to mention that the threshold chosen by hand are arbitrary and that a decision tree will chose the "best" splits based on data.

Also just curious why you switched to 27 years (rather than 25 years). It would be good to mention that these are done by eye and very approximate.

Reply:
25 was letting a bunch of blue point on its right (on the figure). 27 corresponds to  the age were the orange dot start (we can actually mention how we decided this rule). I will make the change

Reply:
I added that this is "one" arbitrary possibility chosen such that we isolate best the two classes with only 2 rules.